lib/shape.ml

```diff
@@ -24,36 +25,59 @@ let draw_line x1 y1 x2 y2 = draw_poly_line [| (x1, y1); (x2, y2) |]
 let denormalize point =
   { x = point.x + canvas_mid.x; y = point.y + canvas_mid.y }
 
-let render_shape s =
+  let render_shape s =
   match s with
   | Circle circle ->
-      draw_circle (denormalize circle.c).x (denormalize circle.c).y
-        circle.radius
+      set_color black;
+      if circle.fill <> Transparent then begin
+        set_color (match circle.fill with RGB (r, g, b) -> rgb r g b | Transparent -> black);
+        fill_circle (denormalize circle.c).x (denormalize circle.c).y circle.radius;
+      end;
+      if circle.stroke <> Transparent then begin
+        set_color (match circle.stroke with RGB (r, g, b) -> rgb r g b | Transparent -> black);
+        draw_circle (denormalize circle.c).x (denormalize circle.c).y circle.radius;
+      end;
```

Comment:
I think this can be simplified a bit with nested match statements. The `Transparent -> black` part is redundant I think, since the previous check makes sure it's not `Transparent`.

Reply:
Okay, I'll try that.